lib/cloud/aws/config/config.go

```diff
@@ -0,0 +1,225 @@
+// assuming AWS IAM Roles.
+func GetAWSConfig(ctx context.Context, region string, opts ...AWSOptionsFn) (aws.Config, error) {
+	var options awsOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
+	if options.baseConfig == nil {
+		cfg, err := getAWSConfigForRegion(ctx, region, options)
+		if err != nil {
+			return aws.Config{}, trace.Wrap(err)
+		}
+		options.baseConfig = &cfg
+	}
+	if options.assumeRoleARN == "" {
+		return *options.baseConfig, nil
+	}
+	return getAWSConfigForRole(ctx, region, options)
+}
+
+// awsAmbientConfigProvider loads a new config using the environment variables.
+func awsAmbientConfigProvider(region string, cred aws.CredentialsProvider, options awsOptions) (aws.Config, error) {
+	opts := buildAWSConfigOptions(region, cred, options)
+	cfg, err := config.LoadDefaultConfig(context.Background(), opts...)
+	return cfg, trace.Wrap(err)
+}
+
+func buildAWSConfigOptions(region string, cred aws.CredentialsProvider, options awsOptions) []func(*config.LoadOptions) error {
+	opts := []func(*config.LoadOptions) error{
+		config.WithRegion(region),
+		config.WithCredentialsProvider(cred),
+	}
+	if modules.GetModules().IsBoringBinary() {
+		opts = append(opts, config.WithUseFIPSEndpoint(aws.FIPSEndpointStateEnabled))
+	}
+	if options.customRetryer != nil {
+		opts = append(opts, config.WithRetryer(options.customRetryer))
+	}
+	if options.maxRetries != nil {
+		opts = append(opts, config.WithRetryMaxAttempts(*options.maxRetries))
+	}
+	return opts
+}
+
+// getAWSConfigForRegion returns AWS config for the specified region.
+func getAWSConfigForRegion(ctx context.Context, region string, options awsOptions) (aws.Config, error) {
+	if err := options.checkAndSetDefaults(); err != nil {
+		return aws.Config{}, trace.Wrap(err)
+	}
+
+	var cred aws.CredentialsProvider
+	if options.credentialsSource == cloud.CredentialsSourceIntegration {
+		if options.awsIntegrationCredentialsProvider == nil {
+			return aws.Config{}, trace.BadParameter("missing aws integration credential provider")
+		}
+
+		slog.DebugContext(ctx, "Initializing AWS config with integration.", "region", region, "integration", options.integration)
+		var err error
+		cred, err = options.awsIntegrationCredentialsProvider(ctx, region, options.integration)
+		if err != nil {
+			return aws.Config{}, trace.Wrap(err)
+		}
+	} else {
+		slog.DebugContext(ctx, "Initializing AWS config from environment.", "region", region)
+	}
+
+	cfg, err := awsAmbientConfigProvider(region, cred, options)
+	return cfg, trace.Wrap(err)
+}
+
+// getAWSConfigForRole returns an AWS config for the specified region and role.
+func getAWSConfigForRole(ctx context.Context, region string, options awsOptions) (aws.Config, error) {
+	if err := options.checkAndSetDefaults(); err != nil {
+		return aws.Config{}, trace.Wrap(err)
+	}
+
+	stsClient := sts.NewFromConfig(*options.baseConfig)
+	cred := stscreds.NewAssumeRoleProvider(stsClient, options.assumeRoleARN, func(aro *stscreds.AssumeRoleOptions) {
```

Comment:
IIRC, the SDKv2 assume role will fail if region is empty, but it was fine in SDKv1.  Could we double-check that? 

There are a few occasions for STS/IAM clients that we use empty region. The workaround could be defaulting to `us-east-1` when region is not provided by our code or the default config.
https://github.com/gravitational/teleport/pull/47898#discussion_r1815511662